test/CMakeLists.txt

```diff
@@ -1,7 +1,13 @@
 add_custom_target (tests)
 
+if(BUILD_TESTS)
+    set(TESTING_FLAG  "")
+else()
+    set(TESTING_FLAG EXCLUDE_FROM_ALL)
+endif()
+
```

Comment:
wouldn't be easier to exlude the compiled tests in all cases and provide a custom target ?